config/confd/templates/docker-registry.yml.tmpl

```diff
@@ -3,52 +3,55 @@ log:
   level: warn
 http:
   addr: :80
-  host: https://{{getv "/resin/registry2/host"}}
-  secret: {{getv "/resin/registry2/secret-key"}}
+  host: https://{{getenv "BALENA_REGISTRY2_HOST"}}
+  secret: {{getenv "REGISTRY2_SECRETKEY"}}
   headers:
     X-Content-Type-Options: [nosniff]
 auth:
   token:
-    service: {{getv "/resin/registry2/host"}}
-    realm: {{getv "/resin/api/tokenauth/realm"}}
-    issuer: {{getv "/resin/api/tokenauth/issuer"}}
+    service: {{getenv "BALENA_REGISTRY2_HOST"}}
+    realm: {{getenv "BALENA_TOKEN_AUTH_REALM"}}
+    issuer: {{getenv "BALENA_TOKEN_AUTH_ISSUER"}}
     rootcertbundle: /certs/registry-tokenauth.crt
 storage:
-{{if eq (getv "/resin/registry2/cache-enabled" "false") "true"}}
+{{if eq (getenv "REGISTRY2_CACHE_ENABLED" "false") "true"}}
   cache:
     blobdescriptor: redis
 {{end}}
-{{if exists "/resin/registry2/s3/bucket"}}
+{{if getenv "REGISTRY2_S3_BUCKET"}}
   s3:
-    accesskey: {{getv "/resin/registry2/s3/key"}}
-    secretkey: {{getv "/resin/registry2/s3/secret"}}
-    region: {{getv "/resin/common/region"}}
-{{if exists "/resin/registry2/s3/regionendpoint"}}
-    regionendpoint: {{getv "/resin/registry2/s3/regionendpoint"}}
+    accesskey: {{getenv "REGISTRY2_S3_KEY"}}
+    secretkey: {{getenv "REGISTRY2_S3_SECRET"}}
+    region: {{getenv "COMMON_REGION"}}
+{{if getenv "REGISTRY2_S3_REGION_ENDPOINT"}}
+    regionendpoint: {{getenv "REGISTRY2_S3_REGION_ENDPOINT"}}
 {{end}}
-    bucket: {{getv "/resin/registry2/s3/bucket"}}
+    bucket: {{getenv "REGISTRY2_S3_BUCKET"}}
     encrypt: false
     secure: true
     v4auth: true
-    rootdirectory: {{getv "/resin/registry2/storage-path"}}
+    rootdirectory: {{getenv "REGISTRY2_STORAGEPATH"}}
 {{else}}
   filesystem:
-    rootdirectory: {{getv "/resin/registry2/storage-path"}}
+    rootdirectory: {{getenv "REGISTRY2_STORAGEPATH"}}
 {{end}}
 
-{{if exists "/resin/registry2/s3/cloudfront/endpoint"}}
+{{if getenv "REGISTRY2_S3_CLOUDFRONT_ENDPOINT"}}
 middleware:
   storage:
     - name: cloudfront
       options:
         ipfilteredby: none
-        baseurl: https://{{getv "/resin/registry2/s3/cloudfront/endpoint"}}/
+        baseurl: https://{{getenv "REGISTRY2_S3_CLOUDFRONT_ENDPOINT"}}/
         privatekey: /etc/ssl/private/cloudfront-pk.pem
-        keypairid: {{getv "/resin/registry2/s3/cloudfront/keypairid"}}
+        keypairid: {{getenv "REGISTRY2_S3_CLOUDFRONT_KEYPAIR_ID"}}
 {{end}}
 
-{{if eq (getv "/resin/registry2/cache-enabled" "false") "true"}}
+{{if eq (getenv "REGISTRY2_CACHE_ENABLED" "false") "true"}}
 redis:
-  addr: {{getv "/resin/registry2/cache-addr" "127.0.0.1:6379"}}
-  db: {{getv "/resin/registry2/cache-db" "0"}}
+  addr: {{getenv "REGISTRY2_CACHE_ADDR" "127.0.0.1:6379"}}
+  db: {{getenv "REGISTRY2_CACHE_DB" "0"}}
 {{end}}
+
+redirect:
+  disable: true
```

Comment:
This directive is invalid and it was already dropped in the etcd-based conf but apparently forgotten in the env-based one. Let's remove it.

Reply:
fixed